### PR TITLE
Fixes #22276 - Don't pass AC::Params into actions

### PR DESCRIPTION
**Pull request by jturel**
Here's how we might handle strong parameters since ActionController::Parameters is not a superset of the Hash API.

- Call .to_h on permitted params
- Call .to_unsafe_h on params with nested hashes otherwise we need to disable strong params or permit deep into the structure
- Don't pass actioncontroller::params beyond controllers - it's risky.

### CODE REVIEW

File: app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb [hunk 1]
```diff
@@ -417,7 +417,7 @@ def get_content_view_environments(label = nil, organization = nil)
     end
 
     def rhsm_params
-      params.slice(:name, :type, :facts, :installedProducts, :autoheal, :releaseVer, :serviceLevel, :uuid, :capabilities, :guestIds, :lastCheckin)
+      params.slice(:name, :type, :facts, :installedProducts, :autoheal, :releaseVer, :serviceLevel, :uuid, :capabilities, :guestIds, :lastCheckin).to_h
```
Comment (jturel) [inline app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb]: This works w/o .permit b/c strong params are disabled in this controller

File: test/controllers/api/v2/package_groups_controller_test.rb [hunk 2]
```diff
@@ -119,7 +119,6 @@ def test_create_and_delete
       parameters = { :repository_id => @repo.id, :name => 'My_Group', :description => "My Group", :mandatory_package_names => ["katello-agent"]}
       assert_sync_task(::Actions::Katello::Repository::UploadPackageGroup) do |repository, params|
         repository.must_equal @repo
-        params[:repository_id].to_i.must_equal @repo.id.to_i
```
Comment (jturel) [inline test/controllers/api/v2/package_groups_controller_test.rb]: repository_id isn't being used by UploadPackageGroup action.

**Comment by jturel** [comment]
[test rails-5]

**Comment by danseethaler** [comment]
@jturel I'm guessing this was closed on accident?

**Comment by jturel** [comment]
@danseethaler kind of. there's a new one with these fixes and more: https://github.com/Katello/katello/pull/7160

**Comment by danseethaler** [comment]
@jturel ah perfect - thanks.

**Pull request closed without merging**